src/main/scala/org/apache/mesos/chronos/scheduler/api/DependentJobResource.scala

```diff
@@ -57,10 +57,16 @@ class DependentJobResource @Inject()(
         val parents = jobGraph.parentJobs(newJob)
         oldJob match {
           case j: DependencyBasedJob =>
-            val oldParents = jobGraph.parentJobs(j)
-            oldParents.map(x => jobGraph.removeDependency(x.name, oldJob.name))
+            val newParentNames = parents.map(_.name)
+            val oldParentNames = jobGraph.parentJobs(j).map(_.name)
+
+            if (newParentNames != oldParentNames) {
+              oldParentNames.foreach(jobGraph.removeDependency(_, oldJob.name))
+              newParentNames.foreach(jobGraph.addDependency(_, newJob.name))
+            }
             jobScheduler.removeSchedule(j)
           case j: ScheduleBasedJob =>
```

Comment:
Ah, understood! Thanks for the explanation, and I did a further review. We are updating parent dependencies only (and this is only the case of dependencybasedjobs and scheduledjobs which have become dependencybasedjobs)  and we don't want to get the counts out of sync when updating the graph for the dependency based job. We should be good to go.